src/syft/lib/python/set.py

```diff
@@ -36,6 +37,13 @@ def id(self) -> UID:
         """
         return self._id
 
+    def upcast(self) -> set:
```

Comment:
I can't find TypedSet, so ill just use typing.Set here. Looks like in Python 3.9 this will actually be going back to the older style builtins which is nice. Also I assume that not setting a type would be the equivalent of Any.

https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

```
# For collections, the type of the collection item is in brackets
# (Python 3.9+)
x: list[int] = [1]
x: set[int] = {6, 7}

# In Python 3.8 and earlier, the name of the collection type is
# capitalized, and the type is imported from 'typing'
x: List[int] = [1]
x: Set[int] = {6, 7}
```